[TSK-633] [TSK-683] [TSK-684] feat(ui-kit): Notification and Activity History component is added wi…

manishRise11 opened the pull request:
## Description

<!-- Provide a clear and concise description of the changes made in this pull request for UI-Kit components. -->

- Component name: `Notification` , `Activity history`
- Figma link or screenshot: [Figma Design](https://www.figma.com/)

![Image](https://github.com/user-attachments/assets/f3806610-27a0-4ef2-8ad8-efeeaf52c782)

![Image](https://github.com/user-attachments/assets/d276d370-c8ed-460e-9d90-446aef5abb73)


---

## Checklist

- [x] I have run the lint, build, and test commands locally, as explained in the README, before creating this PR to ensure the code is clean and functional.
- [x] My changes align with the **core models** that have been discussed and agreed upon.
- [x] The component design has been compared to the **Figma design** to ensure consistency.
- [x] The **component is fully responsive** and behaves as expected across different screen sizes.
- [x] The component **supports keyboard web accessibility** (if applicable).
- [x] I have written **unit tests** or **integration tests** to cover the component's functionality above the coverage threshold.
- [x] I have updated the **documentation** for the component, including usage examples and guidelines.
- [x] I have **commented** my code, particularly in complex or non-obvious areas.
- [x] The **component is properly exported** for use in other parts of the application.

---

## Related Issues

<!-- Link any related issue or task. For example: -->
Fixes #48

manishRise11 commented:
@alebg  sir, I have changed this to incorporate issue #125.

- Changed the initial state of `visibleCount` to `null` to allow dynamic initialization
- Added a `useEffect` hook to update `visibleCount` with the value of `maxActivities` whenever it changes.

carlapbg reviewed (changes requested):
Great job! Just one thing: the "Mark all as read" button should be using the variant primary. Right now is using text

alebg commented on packages/ui-kit/lib/components/notifications/activity.tsx:
```diff
@@ -0,0 +1,229 @@
+  return (
+    <>
+      {layout === 'vertical' && children && <>{children}</>}
+      <a href={action.url} className="block">
```
hi @manishRise11 ! could you please use a callback passed from the outside that takes as input `action.url`, instead of hard-coding this href?

alebg commented on packages/ui-kit/lib/components/notifications/recent-activity.tsx:
```diff
@@ -0,0 +1,108 @@
+  variation='Pop-up'
+}) => {
+  const dictionary = getDictionary(locale);
+  const [visibleCount, setVisibleCount] = useState<number>(null);
```
@manishRise11 since `maxActivities` is sort of an initial value (the major exception for the props in state), you can put it in this state. Otherwise it's a bit strange to reason about that the initial state of the amount of activities shown is 0 / null

alebg commented on packages/ui-kit/lib/components/notifications/recent-activity.tsx:
```diff
@@ -0,0 +1,108 @@
+        {displayedActivities.map((activity, index) => (
+          <Activity key={index} {...activity}  locale={locale}/>
+        ))}
+        {filteredActivities.length > maxActivities &&
```
@manishRise11 seems to me that the condition `filteredActivities.length > maxActivities &&` is redundant if you have `visibleCount < filteredActivities.length` already. In the latter, you already capture the case in which the length is higher than maxActivities before the user clicks and triggers handleViewAll. And after the user clicks on it, then it just goes away because `x < x` is always false

Correct me if i'm wrong though.

alebg reviewed (changes requested):
@manishRise11 thanks a lot. Looks mostly good, but please address my inline comments.

Additionally, please try to apply the children as props strategy in the recent activity component. I.e., making it more like a layout component. Otherwise we may have unnecessary re-renders every time there's a new activity

ObitoDevv commented:
@alebg Hi Sir, we have done changes as per the requirements. Can you please review it ?

alebg commented on packages/ui-kit/lib/components/notifications/recent-activity.tsx:
```diff
@@ -0,0 +1,143 @@
+  onClickMarkAllAsRead?: () => void;
+  onViewAll?: () => void;
+  className?: string;
+  variation?: 'Pop-up' | 'Feed' | 'Search';
```
TBD: search variation could be its own thing, implemented with AG Grid

alebg commented on packages/ui-kit/lib/components/notifications/recent-activity.tsx:
```diff
@@ -0,0 +1,143 @@
+  onClickMarkAllAsRead,
+  onViewAll,
+  className,
+  variation = 'Pop-up',
```
@manishRise11 please remove this default value and make 'variation' mandatory

alebg commented on packages/ui-kit/lib/components/notifications/recent-activity.tsx:
```diff
@@ -0,0 +1,143 @@
+          )}
+        </div>
+      )}
+      {variation === 'Search' && (
```
TBD: this might have to be moved to its own component

alebg reviewed (changes requested):
@manishRise11 please remove the "search" variation from this component, as we will implement it with AG Grid in a different task and PR: #134 

You can keep it in the git history (or in a branch of your own) so you can refer to it for the styling, etc.

manishRise11 commented:
@alebg sir, I have updated the code as per your instructions.

- Removed 'Search' from the variation.
- Removed the default value of the variation and made the 'variation' field mandatory.
- Implemented a callback function for the 'View all...' button.

alebg commented:
Hey @manishRise11 , I can't get any of the callbacks to work (no alerts anywhere), is it working for you?

alebg commented on packages/ui-kit/stories/notifications/activity.stories.tsx:
```diff
@@ -76,7 +77,9 @@ export const VerticalLayout: Story = {
     ...Default.args,
     layout: 'vertical',
-    platformName: '',
+    platformName: '2028-08-07T21:17:00.000Z',
```
@manishRise11 could you make the example more realistic please? this doesn't look like a platform name

manishRise11 commented:
@alebg Sir, I have updated the platform name as per your suggestion to make it more realistic. Additionally, I verified that the callbacks are working fine on my device across all three browsers: Safari, Brave, and Chrome.
<img width="905" alt="Screenshot 2025-04-10 at 9 29 48 AM" src="https://github.com/user-attachments/assets/8e8e5471-08e4-40e9-b1b6-bf7b0f59552c" />
<img width="757" alt="Screenshot 2025-04-10 at 9 30 22 AM" src="https://github.com/user-attachments/assets/131de137-c88b-42c7-87ff-372e1944dbde" />
<img width="821" alt="Screenshot 2025-04-10 at 9 30 48 AM" src="https://github.com/user-attachments/assets/035155de-cae9-427d-942a-74ce87d2aa6b" />
<img width="821" alt="Screenshot 2025-04-10 at 9 31 00 AM" src="https://github.com/user-attachments/assets/71a2500c-d437-4473-99d1-3fc03f6a669f" />

carlapbg reviewed (changes requested):
Hi! Just a couple of things:

- Can you add a gap between the notification circle and the text?
![Screenshot 2025-04-15 at 15 25 49](https://github.com/user-attachments/assets/6eccb947-2761-487b-ac7f-65cd5c88399b)


- Also, I think the circle icon is too big. Right now is w-3 h-3, please change it to w-2 h-2

Thanks!

alebg commented on packages/ui-kit/lib/components/notifications/recent-activity.tsx:
```diff
@@ -0,0 +1,89 @@
+import React, { FC, useState, ReactNode } from 'react';
```
@manishRise11 please remove all unused imports. Crucially, if they are client-side react hooks (useX), this will turn this component into a client-side component. Which, as discussed many times over, is *not* what we want for layout-like components, we want them to be server-side.

manishRise11 replied:
@alebg sir, I have removed all unnecessary imports.

alebg reviewed (changes requested):
@manishRise11 please address my inline comment and @carlapbg 's comment
